test/boost/continuous_data_consumer_test.cc

```diff
@@ -118,3 +119,82 @@ SEASTAR_THREAD_TEST_CASE(test_read_unsigned_vint) {
     }
 }
 
+class skipping_consumer final : public data_consumer::continuous_data_consumer<skipping_consumer> {
+    int _initial_data_size;
+    int _to_skip;
+    int _next_data_size;
+    processing_result_generator _gen;
+    temporary_buffer<char>* _processing_data;
+
+    // stream starting with initial_data_size 'a's, followed by to_skip 'b's,
+    // ending with next_data_size 'a's, returning randomly sized chunks
```

Comment:
Is it really randomly-sized chunks? Looks like 1-byte chunks below, no?

Reply:
My bad, at some point I thought I should maybe use random size of chunks to test additional cases, but in the end I decided that it would only distract the reader from the main case that we're testing, I've just uploaded a fix